packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,264 @@
+      >}-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['weight'], 'regular'>>}`
+    | `body-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['size'], 'xsmall'>>}`
+    | `body-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['size'], 'xsmall'>>}-${NonNullable<
+        Exclude<TextProps<{ variant: 'body' }>['weight'], 'regular'>
+      >}`;
+  /**
+   * Indicates what the suffix of amount should be
+   *
+   * @default 'decimals'
+   */
+  suffix?: 'decimals' | 'none' | 'humanize';
+  /**
+   * Makes the prefix symbol and decimal digits small and faded
+   *
+   * @default true
+   */
+  isAffixSubtle?: true | false;
+  /**
+   * Prefix to be shown before the amount value. The prefix can be either a currency symbol or a currency code.
+   *
+   * @default 'currency-symbol'
+   */
+  prefix?: 'currency-symbol' | 'currency-code';
+  /**
+   * The currency of the amount.
+   *
+   * @default 'INR'
+   * */
+  currency?: Currency;
+} & TestID &
+  StyledPropsBlade;
+
+type ColorProps = {
+  amountValueColor: BaseTextProps['color'];
+  affixColor: BaseTextProps['color'];
+};
+
+const getTextColorProps = ({ intent }: { intent: AmountProps['intent'] }): ColorProps => {
+  const props: ColorProps = {
+    amountValueColor: 'surface.text.normal.lowContrast',
+    affixColor: 'surface.text.muted.lowContrast',
+  };
+  if (!intent && intent !== 'neutral') return props;
+  props.amountValueColor = `feedback.text.${intent}.lowContrast`;
+  props.affixColor = `feedback.text.${intent}.lowContrast`;
+  return props;
+};
+
+interface AmountValue extends Omit<AmountProps, 'value'> {
+  affixColor: BaseTextProps['color'];
+  amountValueColor: BaseTextProps['color'];
+  value: string;
+  size: Exclude<AmountProps['size'], undefined>;
+}
+
+const AmountValue = ({
+  value,
+  size,
+  amountValueColor,
+  isAffixSubtle,
+  suffix,
+  affixColor,
+}: AmountValue): ReactElement => {
+  const affixFontWeight = isAffixSubtle ? 'regular' : 'bold';
+  const affixFontSize = isAffixSubtle ? affixFontSizes[size] : amountFontSizes[size];
+  const valueForWeight = size.includes('bold') || size.startsWith('title') ? 'bold' : 'regular';
+  if (suffix === 'decimals' && isAffixSubtle) {
+    const integer = value.split('.')[0];
+    const decimal = value.split('.')[1];
+
+    // Native does not support alignItems of Text inside a div, insted we need to wrap is in a Text
+    const AmountWrapper = getPlatformType() === 'react-native' ? BaseText : React.Fragment;
+
+    return (
+      <AmountWrapper>
+        <BaseText
+          fontSize={amountFontSizes[size]}
+          fontWeight={valueForWeight}
+          color={amountValueColor}
+        >
+          {integer}.
+        </BaseText>
+        <BaseText
+          marginLeft="spacing.1"
+          fontWeight={affixFontWeight}
+          fontSize={affixFontSize}
+          color={affixColor}
+        >
+          {decimal || '00'}
+        </BaseText>
```

Comment:
Could you please check if this doesn't cause unexpected wrapping? For example where the decimal might wrap to a different line in case integer part is large.

Reply:
No, it doesn't.